src/core/commands.mjs

```diff
@@ -1794,6 +1794,8 @@ export async function SaveImage (sender, data) {
       queryOptions.headers = [ { name: "Referer", value: documentValues.referrer } ];
     }
 
+    // download in incognito window if currently in incognito mode
+    queryOptions.incognito = (await browser.windows.getCurrent()).incognito;
```

Comment:
I think we can simply use `sender.tab.incognito`.
Also can you please add it directly here: https://github.com/Robbendebiene/Gesturefy/blob/43819d5c67c99417407f34ce1baf8ddaa0f1232b/src/core/commands.mjs#L1757-L1759